src/femr/labelers/benchmarks.py

```diff
@@ -0,0 +1,741 @@
+                # and only until the visit ends (`e.end`) or a death event
+                #   occurs (`earliest_death_time`)
+                end_of_stay: datetime.datetime = min(e.end, earliest_death_time)
+                event_time = e.start + datetime.timedelta(hours=4)
+                while event_time < end_of_stay:
+                    los: float = (end_of_stay - event_time).total_seconds() / 3600
+                    labels.append(Label(event_time, los))
+                    event_time += datetime.timedelta(hours=1)
+                    assert los >= 0, (
+                        f"LOS should never be negative, but end_of_stay={end_of_stay}"
+                        f" - event_time={event_time} = {end_of_stay - event_time} for"
+                        f" patient {patient.patient_id}"
+                    )
+        return labels
+
+
+##########################################################
+##########################################################
+# Abnormal Lab Value Tasks
+#
+# Citation: Few shot EHR benchmark (ours)
+##########################################################
+##########################################################
+
+
+class ThrombocytopeniaInstantLabValueLabeler(InstantLabValueLabeler):
+    """lab-based definition for thrombocytopenia based on platelet count (10^9/L).
+    Thresholds: mild (<150), moderate(<100), severe(<50), and reference range."""
+
+    original_omop_concept_codes = [
+        "LOINC/LP393218-5",
+        "LOINC/LG32892-8",
+        "LOINC/777-3",
+    ]
+
+    def value_to_label(self, raw_value: str, unit: Optional[str]) -> str:
+        if raw_value.lower() in ["normal", "adequate"]:
+            return "normal"
+        value = float(raw_value)
+        if value < 50:
+            return "severe"
+        elif value < 100:
+            return "moderate"
+        elif value < 150:
+            return "mild"
+        return "normal"
+
+
+class HyperkalemiaInstantLabValueLabeler(InstantLabValueLabeler):
+    """lab-based definition for hyperkalemia using blood potassium concentration (mmol/L).
+    Thresholds: mild(>5.5),moderate(>6),severe(>7), and abnormal range."""
+
+    original_omop_concept_codes = [
+        "LOINC/LG7931-1",
+        "LOINC/LP386618-5",
+        "LOINC/LG10990-6",
+        "LOINC/6298-4",
+        "LOINC/2823-3",
+    ]
+
+    def value_to_label(self, raw_value: str, unit: Optional[str]) -> str:
+        if raw_value.lower() in ["normal", "adequate"]:
+            return "normal"
+        value = float(raw_value)
+        if unit is not None:
+            unit = unit.lower()
+            if unit.startswith("mmol/l"):
+                # mmol/L
+                # Original OMOP concept ID: 8753
+                value = value
+            elif unit.startswith("meq/l"):
+                # mEq/L (1-to-1 -> mmol/L)
+                # Original OMOP concept ID: 9557
+                value = value
+            elif unit.startswith("mg/dl"):
```

Comment:
Hmm. I think we need to handle these unit mappings ideally in the ETL layer. How often do you notice this triggers?

Reply:
Can we keep these in for now for the sake of reproducibility? But yes agreed going forward it would be ideal to move to ETL.